Src/MyCameraPlugin.cs

```diff
@@ -124,31 +135,139 @@ public void OnFixedUpdate()
+                    Log("Done Loading Settings");
+                    Log("New Camera Count: " + CameraPluginSettings.CameraDataList.Count);
+                }
+            }
+            else
+            {
+                // We're still in the menu so reset the timer
+                _elapsedTime = 0.0f;
+                nextChangeTimer = 1.0f;
+            }
+        }
+        else
+        {
+            if (!inGame)
+            {
+                Log("Transitioning to inGame");
+                inGame = true;
+                _elapsedTime = 2.0f;
+                nextChangeTimer = 1.0f;
+
+                if (useHttpStatus)
+                {
+                    // If HTTPStatus is available, see if we can extract some of the song details
+                    Log("Song Name: " + BS.songName);
+                    Log("Song SubName: " + BS.songSubName);
+                    Log("Song AuthorName: " + BS.songAuthorName);
+                    Log("Level Author: " + BS.levelAuthorName);
+                    Log("Song Hash: " + BS.songHash);
+                    Log("Level ID: " + BS.levelId);
+
+                    // We need to check to see if there is a song-specific settings file, and if so, load that instead
+                    Log("Checking to see if there are song-specific Settings");
+                    CameraPluginSettings.LoadSettings("settings." + BS.songHash + ".txt");
+
+                    if (CameraPluginSettings.SongSpecific)
+                    {
+                        Log("Song specific settings found!");
+                        Log("New Camera Count: " + CameraPluginSettings.CameraDataList.Count);
+                    }
+                    else
+                        Log("No song specific settings file found");
+                }
+            }
+        }
+
+        if (useHttpStatus && BS.paused)
+        {
+            //We're paused, don't do anything
+        }
+        else if (_elapsedTime >= nextChangeTimer || transitionToMenu)
         {
             cameraLerpValue = 0.02f;
             Log("New Camera Selection started");
 
             var newCameraIndex = currentCameraIndex;
 
-            // Don't pick the same camera again for a few times
-            while (previousCameraIndices.Contains(newCameraIndex) || (currentCameraType == CameraType.Orbital && CameraPluginSettings.CameraDataList[newCameraIndex].Type == CameraType.Orbital))
+            if (transitionToMenu)
             {
-                newCameraIndex = rand.Next() % CameraPluginSettings.CameraDataList.Count;
+                transitionToMenu = false;
+                currentCameraIndex = 0;
+                
+                //currentCameraData = CameraPluginSettings.CameraDataList[currentCameraIndex];
+                //currentCameraType = currentCameraData.Type;
+
+                // I hate that this is hardcoded but unless we add a specific 'menu' camera in the base settings file it'll have to do for now
```

Comment:
I thought I saw that you had added a menu camera to the settings?

Reply:
Yeah, but it needs some TLC, I like the following idea!